### PR TITLE
Add smi collector

### PR DESCRIPTION
**Pull request by johnsonshi**
Add smi collector for periscope.

For each SMI resource within the cluster,
(1) write the SMI resource details as a table to a TSV file,
(2) write the SMI resource configs to a JSON file.

image

image

### CODE REVIEW

File: deployment/aks-periscope.yaml [hunk 1]
```diff
@@ -23,6 +23,18 @@ rules:
...
+  verbs: ["get", "list", "watch"]
+- apiGroups: ["split.smi-spec.io"]
+  resources: ["trafficsplits"]
+  verbs: ["get", "list", "watch"]
+- apiGroups: ["access.smi-spec.io"]
+  resources: ["traffictargets"]
+  verbs: ["get", "list", "watch"]
+- apiGroups: ["specs.smi-spec.io"]
+  resources: ["httproutegroups", "tcproutes", "udproutes"]
+  verbs: ["get", "list", "watch"]
```
Comment (johnsonshi) [inline deployment/aks-periscope.yaml]: These kubernetes rbac `ClusterRole` configs allow the role to access resources within the SMI api groups (`*.smi-spec.io`).

Is there a way to specify rbac api group access using wildcards? I'm trying to achieve the following in lieu of what I've written above, but I do not know if this is possible (or even what the syntax should be). Help appreciated.

```yaml
- apiGroups: ["*.smi-spec.io"]
  resources: ["*"]
  verbs: ["get", "list", "watch"]
```

For context on the SMI crds that I am interested in, this is the output when I run `kubectl api-resources | grep smi-spec.io`:

```tsv
NAME                              SHORTNAMES   APIVERSION                          NAMESPACED   KIND
traffictargets                    tt           access.smi-spec.io/v1alpha3         true         TrafficTarget
httproutegroups                   htr          specs.smi-spec.io/v1alpha4          true         HTTPRouteGroup
tcproutes                         tr           specs.smi-spec.io/v1alpha4          true         TCPRoute
udproutes                         ur           specs.smi-spec.io/v1alpha4          true         UDPRoute
trafficsplits                     ts           split.smi-spec.io/v1alpha2          true         TrafficSplit
```

Comment (ritazh) [reply deployment/aks-periscope.yaml]: Not for apiGroups, according to this: https://github.com/kubernetes/kubernetes/blob/a55bd631728590045b51a4f65bba31aed1415571/staging/src/k8s.io/api/rbac/v1beta1/types.go#L53-L56
You can try: 
```yaml
- apiGroups: ["access.smi-spec.io", "specs.smi-spec.io"]
  resources: ["*"]
  verbs: ["get", "list", "watch"]
```

Comment (johnsonshi) [reply deployment/aks-periscope.yaml]: Thanks Rita!

File: pkg/collector/collector.go [hunk 2]
```diff
@@ -69,3 +76,26 @@ func (b *BaseCollector) Export() error {
 
 	return nil
 }
+
+// Helper function to collect output of a given kubectl command to a file. Returns kubectl's stderr output if stdout output is empty.
+func (b *BaseCollector) collectResourceToFileUsingKubectl(rootPath, fileName string, kubeCmds []string) error {
```
Comment (SanyaKochhar) [inline pkg/collector/collector.go]: can this name be simplified to something shorter like `collectKubeResourceToFile` ?

**Review by SanyaKochhar — APPROVED** [review]
This looks good. I was just wondering whether the collector should be renamed to something else because technically no "logs" are being collected, only SMI resource details.

**Pull request merged**